__tests__/app.test.js

```diff
@@ -0,0 +1,67 @@
+const request = require(`supertest`);
+const app = require(`../app.js`);
+const seed = require(`../db/seeds/seed.js`);
+const db = require(`../db/connection.js`);
+const testData = require(`../db/data/test-data/index.js`)
+const { TestWatcher } = require("jest");
+
+beforeEach(() => {
+    return seed(testData)
+})
+
+afterAll(() => {
+    return db.end()
+})
+
+describe (`/api/topics`, () => {
+    test(`GET - 200: returns an array of objects`, () => {
+        return request(app)
+            .get(`/api/topics`)
+            .expect(200)
+            .then((res) => {
+                expect(Array.isArray(res.body)).toEqual(true);
+            })
+    })
+
+    test(`GET - 200: objects contain keys for slug and description`, () => {
+        return request(app)
+            .get(`/api/topics`)
+            .expect(200)
+            .then((res) => {
+                res.body.forEach((topic) => {
+                    expect(topic).toMatchObject({
+                        slug: expect.any(String),
+                        description: expect.any(String)
+                    })
+                })
+            })
+    })
```

Comment:
This test is sound! But a couple of alterations are required...
- You will need to assert the length of the array - otherwise the forEach will never run
- We don't want the response body to be the array, but instead an object with a key of `topics` - on that key is the array

Reply:
I believe the length is asserted, though I'm not sure I did it as cleanly as is possible. Not sure how to get the response body into an object while still having it be useful - looking at the ticket, it seems to just want an array..? Leaving this open as uncertain.